mlflow/gateway/providers/cohere.py

```diff
@@ -174,6 +179,144 @@ def embeddings_to_model(cls, payload, config):
                 )
         return rename_payload_keys(payload, key_mapping)
 
+    @classmethod
+    def chat_to_model(cls, payload, config):
+        if payload["n"] != 1:
+            raise HTTPException(
+                status_code=422,
+                detail=f"Parameter n must be 1 for Cohere chat, got {payload['n']}.",
+            )
+        del payload["n"]
+
+        if "stop" in payload:
+            raise HTTPException(
+                status_code=422,
+                detail="Parameter stop is not supported for Cohere chat.",
+            )
+        payload = cls._scale_temperature(payload)
+
+        last_message = payload["messages"][-1]
+        if last_message["role"] != "user":
+            raise HTTPException(
+                status_code=422,
+                detail=f"Last message must be from user, got {last_message['role']}.",
+            )
+        payload["message"] = last_message["content"]
+
+        if len(payload["messages"]) > 1:
+            payload["chat_history"] = [
+                {
+                    "role": "USER" if m["role"] == "user" else "CHATBOT",
+                    "message": m["content"],
+                }
+                for m in payload["messages"][:-1]
+            ]
+        del payload["messages"]
```

Comment:
can we use `if messages := payload.pop("messages", None)` instead?

Reply:
i don't think `del` either, let me update it